aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+        assert!(exists<WhitelistMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        // cannot update whitelisted addresses if the whitelist minting period has already passed
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        assert!(whitelist_mint_config.whitelist_minting_end_time > timestamp::now_seconds(), error::permission_denied(EINVALID_UPDATE_AFTER_MINTING));
+
+        let i = 0;
+        while (i < vector::length(&wl_addresses)) {
+            let addr = *vector::borrow(&wl_addresses, i);
+            // assert that the specified address exists
+            assert!(account::exists_at(addr), error::invalid_argument(EACCOUNT_DOES_NOT_EXIST));
+            bucket_table::add(&mut whitelist_mint_config.whitelisted_address, addr, mint_limit);
+            i = i + 1;
+        };
+    }
+
+    /// Add destination tokens.
+    public entry fun add_tokens(
+        admin: &signer,
+        token_uris: vector<String>,
+        property_keys: vector<vector<String>>,
+        property_values: vector<vector<vector<u8>>>,
+        property_types: vector<vector<String>>
+    ) acquires NFTMintConfig, CollectionConfig, WhitelistMintConfig {
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
+
+        // cannot add more token uris if minting has already started
+        assert!(exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        assert!(whitelist_mint_config.whitelist_minting_start_time > timestamp::now_seconds(), error::permission_denied(EINVALID_UPDATE_AFTER_MINTING));
+
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        assert!(vector::length(&token_uris) == vector::length(&property_keys) && vector::length(&property_keys) == vector::length(&property_values) && vector::length(&property_values) == vector::length(&property_types), error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+
+        assert!(vector::length(&token_uris) + vector::length(&token_config.token_uri) <= token_config.collection_maximum || token_config.collection_maximum == 0, error::invalid_argument(EEXCEEDS_COLLECTION_MAXIMUM));
+        vector::append(&mut token_config.token_uri, token_uris);
+        vector::append(&mut token_config.property_keys, property_keys);
+        vector::append(&mut token_config.property_values, property_values);
+        vector::append(&mut token_config.property_types, property_types);
```

Comment:
this won't scale, each one needs to be a distinct field in a table.

Reply:
use a bigvector instead?

Reply:
yup just updated to use a big vector : )

Reply:
okay I get what you meant by the bit vector now... but we still need a way to randomize and find an 0 in the bit vector though? how is this better than the timestamp approach?